test/unit/org/apache/cassandra/inject/ActionBuilder.java

```diff
@@ -127,7 +127,7 @@ public ConditionsBuilder when(Object expression)
 
         public ConditionsBuilder not()
         {
-            if (!(elements.getLast() instanceof LogicOp))
+            if (!elements.isEmpty() && !(elements.getLast() instanceof LogicOp))
```

Comment:
this change seem unrelated?

Reply:
It's necessary for some of the changes in the tests to not thrown a NPE. Namely, if you look at the updated injection conditions in `NodeStartupTest`, they start by a `not()` and so were hitting this (where it was trying to get the last element that just doesn't exists).

Not that this is testing code only.